bigtable-hbase-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAdmin.java

```diff
@@ -0,0 +1,618 @@
+
+import org.apache.hadoop.hbase.ClusterStatus;
+import org.apache.hadoop.hbase.ClusterStatus.Option;
+import org.apache.hadoop.hbase.HColumnDescriptor;
+import org.apache.hadoop.hbase.HTableDescriptor;
+import org.apache.hadoop.hbase.NamespaceDescriptor;
+import org.apache.hadoop.hbase.RegionLoad;
+import org.apache.hadoop.hbase.ServerName;
+import org.apache.hadoop.hbase.TableExistsException;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.TableNotFoundException;
+import org.apache.hadoop.hbase.client.AbstractBigtableAdmin;
+import org.apache.hadoop.hbase.client.AbstractBigtableConnection;
+import org.apache.hadoop.hbase.client.ColumnFamilyDescriptor;
+import org.apache.hadoop.hbase.client.CompactType;
+import org.apache.hadoop.hbase.client.CompactionState;
+import org.apache.hadoop.hbase.client.MasterSwitchType;
+import org.apache.hadoop.hbase.client.RegionInfo;
+import org.apache.hadoop.hbase.client.SnapshotDescription;
+import org.apache.hadoop.hbase.client.SnapshotType;
+import org.apache.hadoop.hbase.client.TableDescriptor;
+import org.apache.hadoop.hbase.client.replication.TableCFs;
+import org.apache.hadoop.hbase.client.security.SecurityCapability;
+import org.apache.hadoop.hbase.quotas.QuotaFilter;
+import org.apache.hadoop.hbase.quotas.QuotaRetriever;
+import org.apache.hadoop.hbase.quotas.QuotaSettings;
+import org.apache.hadoop.hbase.snapshot.HBaseSnapshotException;
+import org.apache.hadoop.hbase.snapshot.RestoreSnapshotException;
+import org.apache.hadoop.hbase.snapshot.SnapshotCreationException;
+import org.apache.hadoop.hbase.snapshot.UnknownSnapshotException;
+
+import com.google.bigtable.admin.v2.ModifyColumnFamiliesRequest.Modification;
+import com.google.cloud.bigtable.hbase.adapters.admin.ColumnDescriptorAdapter;
+
+/**
+ * HBase 2.x specific implementation of {@link AbstractBigtableAdmin}.
+ * @author spollapally
+ */
+@SuppressWarnings("deprecation")
+public class BigtableAdmin extends AbstractBigtableAdmin {
+
+  public BigtableAdmin(AbstractBigtableConnection connection) throws IOException {
+    super(connection);
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public void createTable(TableDescriptor desc) throws IOException {
+    createTable((HTableDescriptor) desc);
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public void createTable(TableDescriptor desc, byte[][] splitKeys) throws IOException {
+    createTable((HTableDescriptor) desc, splitKeys);
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public void createTable(TableDescriptor desc, byte[] startKey, byte[] endKey, int numRegions)
+      throws IOException {
+    createTable((HTableDescriptor) desc, startKey, endKey, numRegions);
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public Future<Void> createTableAsync(TableDescriptor desc, byte[][] splitKeys)
+      throws IOException {
+    createTableAsync((HTableDescriptor) desc, splitKeys);
+    // Consider better options after adding support for async hbase2
```

Comment:
Please add "TODO:" in front of any comments like this, where an action is required.